API/SSW.Rewards.Application/Skills/Commands/DeleteSkill/DeleteSkillCommand.cs

```diff
@@ -25,21 +25,21 @@ public UpsertSkillCommandHandler(ISSWRewardsDbContext context)
 
             public async Task<Unit> Handle(DeleteSkillCommand request, CancellationToken cancellationToken)
             {
-                if (!string.IsNullOrWhiteSpace(request.Skill))
+                if (request?.Id == null)
                 {
-                    var found = _context.Skills.FirstOrDefault(x => x.Name == request.Skill);
+                    throw new Exception("Bad Request");
```

Comment:
I've never been a fan of using exceptions to fail an operation on purpose. If it's a common pattern in this app then ok, but otherwise you should return at least a boolean result to indicate success/failure and then the web layer translate that result into HTTP 404. Also, exceptions are horrible for performance.

Reply:
Yep ok it seems to be a common pattern in this app. Perhaps in the future version we can adopt a better practice of returning result objects instead of throwing exceptions.

Reply:
I think exceptions should be used for truly exceptional circumstances. In this case as the skills list is coming from the API, I would expect someone trying to delete a skill that doesn't exist to be exceptional. However according to the REST standard we should return a 404 if someone tries to delete a resource that doesn't exist, so whether we use an exception here or not, a 500 is the wrong response code, which is what throwing this exception will do.

Reply:
Agreed, I've added a PBI for this: https://github.com/SSWConsulting/SSW.Rewards/issues/265
For this PR I'll keep it as it is unless you want me to change the response code?